pkgs/development/haskell-modules/configuration-nix.nix

```diff
@@ -935,11 +935,11 @@ self: super: builtins.intersectAttrs super {
 
   rel8 = addTestToolDepend super.rel8 pkgs.postgresql;
 
-  cachix = generateOptparseApplicativeCompletion "cachix" (super.cachix.override { nix = pkgs.nix_2_3; });
+  cachix = addBuildDepends (generateOptparseApplicativeCompletion "cachix" (super.cachix.override { nix = pkgs.nix_2_4; })) [ pkgs.nlohmann_json ];
 
-  hercules-ci-agent = super.hercules-ci-agent.override { nix = pkgs.nix_2_3; };
-  hercules-ci-cnix-expr = super.hercules-ci-cnix-expr.override { nix = pkgs.nix_2_3; };
-  hercules-ci-cnix-store = super.hercules-ci-cnix-store.override { nix = pkgs.nix_2_3; };
+  hercules-ci-agent = addBuildDepends (appendConfigureFlag (super.hercules-ci-agent.override { nix = pkgs.nix_2_4; }) "-fnix-2_4") [ pkgs.nlohmann_json ];
+  hercules-ci-cnix-expr = addBuildDepends (appendConfigureFlag (super.hercules-ci-cnix-expr.override { nix = pkgs.nix_2_4; }) "-fnix-2_4") [ pkgs.nlohmann_json ];
+  hercules-ci-cnix-store = addBuildDepends (appendConfigureFlag (super.hercules-ci-cnix-store.override { nix = pkgs.nix_2_4; }) "-fnix-2_4") [ pkgs.nlohmann_json ];
```

Comment:
I'm not against it, but I'm curious why pin to `nix_2_4` instead of just using `nix`?

Reply:
The C++ interface has proved to change significantly enough between "minor" releases. I don't assume it will build with 2.5 and I need to keep track of all this closely anyway, so it's unlikely that I'll forget to bump this, if that's a concern.

Reply:
Got it, makes sense :)